src/SelectPanel/SelectPanel.tsx

```diff
@@ -123,7 +123,7 @@ export function SelectPanel({
       return {
         ...item,
         role: 'option',
-        selected: 'selected' in item && item.selected === undefined ? undefined : finalItemsSelected.includes(item),
+        selected: 'selected' in item && item.selected === undefined ? undefined : !!finalItemsSelected.find(el => el.id === item.id),
```

Comment:
Non-blocking nitpick: `Boolean(finalItemsSelected.find(el => el.id === item.id))` is more readable.

```suggestion
        selected: 'selected' in item && item.selected === undefined ? undefined : Boolean(finalItemsSelected.find(el => el.id === item.id)),
```